python/sidecar/storage.py

```diff
@@ -25,6 +43,29 @@ def load(self):
             ref.chunks = chunks
             self.references.append(ref)
         self.create_corpus()
+
+    def save(self):
+        """
+        Save the references to the storage file as JSON.
+        """
+        contents = [ref.dict() for ref in self.references]
+        with open(self.filepath, 'w') as f:
+            json.dump(contents, f, indent=2)
+    
+    def update(self, target: typing.Reference):
+        """
+        Update a Reference in storage with the target reference.
+        This is used when the client has updated the reference in the UI.
+        """
+        for i, source in enumerate(self.references):
+            if source.filename_md5 == target.filename_md5:
+                logger.info(f"Updating reference {source.source_filename} ({source.filename_md5})")
+                self.references[i] = target
+                break
+        else:
+            logger.info(f"Unable to find reference {target.source_filename} ({target.filename_md5})")
+            return
+        self.save()
```

Comment:
@cguedes @sehyod for now, i'm assuming that you will be passing back in the entire `Reference` object, inclusive of `chunks` and such. it feels a little heavy, but seemed easiest for now. let me know if that's an issue for y'all.

Reply:
alternatively, we could pass in the filename hash (or we should probably use a [UUID](https://docs.python.org/3/library/uuid.html)) and dict, making the function signature something like:
```python
def update(uuid: UUID, updates: dict[string, Any])
```
where `dict[string, Any]` => `field_name_to_update: value_to_set`. would need to think through how to do this for subtypes like Author though. maybe individual methods for `update_reference` and `update_author` to make things clear and specific.

Reply:
I think it is ok, also bc this is JsonStorage file. Maybe we need to expose to the frontend more granular operations (ex: rename citation). So this is ok for now.